src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -188,7 +188,7 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
             if (inputType.ContainsTypeParameter())
             {
                 convertedExpression = expression;
-                if (!hasErrors)
+                if (!hasErrors && expression.ConstantValue is {})
```

Comment:
Why `{}` instead of `object`? If this value becomes a `struct` at some point in the future then this code will continue to compile without any warning.

Reply:
Please add a comment explaining this, perhaps something like `// A non-constant expression will be reported as an error in the caller` if that is the case.

---
In reply to: [278613370](https://github.com/dotnet/roslyn/pull/35249#discussion_r278613370) [](ancestors = 278613370)